src/main/java/com/volasoftware/tinder/services/implementations/FriendServiceImpl.java

```diff
@@ -0,0 +1,74 @@
+package com.volasoftware.tinder.services.implementations;
+
+import com.volasoftware.tinder.constants.AccountConstant;
+import com.volasoftware.tinder.exceptions.AccountNotFoundException;
+import com.volasoftware.tinder.exceptions.FriendExistException;
+import com.volasoftware.tinder.exceptions.FriendNotFoundException;
+import com.volasoftware.tinder.models.Account;
+import com.volasoftware.tinder.repositories.AccountRepository;
+import com.volasoftware.tinder.services.contracts.FriendService;
+import java.util.HashSet;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class FriendServiceImpl implements FriendService {
+
+    private final AccountRepository accountRepository;
+
+    @Override
+    public void addFriend(Long friendId) {
+        Account loggedAccount = getLoggedAccount();
+        Account friendAccount = getAccountById(friendId);
+
+        if (hasFriends(loggedAccount)) {
+            if (friendExist(loggedAccount, friendAccount)) {
+                throw new FriendExistException(AccountConstant.ALREADY_EXIST);
+            }
+        }
+
+        HashSet<Account> friends = new HashSet<>();
+        friends.add(friendAccount);
+        loggedAccount.setFriends(friends);
+        accountRepository.save(loggedAccount);
+    }
+
+    @Override
+    public void removeFriend(Long friendId) {
+        Account loggedAccount = getLoggedAccount();
+        Account friendAccount = getAccountById(friendId);
+
+        if (!hasFriends(loggedAccount)) {
```

Comment:
Is this check pass through an empty collection of friends?

Reply:
There is Logical Not in front of the method and that is creating the confusion, i guess. The code was simplified.